apps/visualization/views.py

```diff
@@ -45,3 +57,99 @@ def get_queryset(self):
             ) for value in result
         ])
         return DataCountryLevelMostRecent.objects.filter(filters).distinct('subvariable')
+
+
+class ExportBaseView(ListAPIView):
+    pagination_class = LimitOffsetPagination
+
+    def get_serializer_class(self):
+        return None
+
+    def validate_export_params_together(self, params):
+        query_params_keys_list = list(params.keys())
+        if not any(elem in ['iso3', 'indicator_id'] for elem in query_params_keys_list):
+            raise ValidationError({'error': 'iso3 or indicator_id params are required'})
+
+        if 'limit' not in query_params_keys_list:
+            raise ValidationError({'error': 'Limit is required.'})
+
+        if 'offset' not in query_params_keys_list:
+            raise ValidationError({'error': 'offset is required.'})
+
+    def process_csv_response(self, header_fields, include_header, qs):
+        response = HttpResponse(
+            content_type='text/csv',
+        )
+        writer = csv.writer(response)
+
+        if include_header:
+            writer.writerow([field for field in header_fields])
+
+        for item in qs:
+            row = [getattr(item, field) for field in header_fields]
+            writer.writerow(row)
+
+        return response
+
+
+class ExportRawDataView(ExportBaseView):
+
+    filterset_class = DataGranularPublicExportFilter
+    queryset = DataGranularPublic.objects.all()
+
+    def get(self, request, format=None):
+        """
+        Export raw data
+        """
+
+        self.validate_export_params_together(self.request.query_params)
+        filterset = DataGranularPublicExportFilter(
+            data=self.request.query_params,
+            queryset=self.get_queryset(),
+        )
+        if filterset.is_valid():
+            header_fields = [field.name for field in DataGranularPublic._meta.get_fields()]
+            include_header = filterset.form.cleaned_data.get('include_header')
```

Comment:
@thenav56  If we get include_header form `self.request.query_params` we have to manually CAST bools,
 If we use `filterset.form.cleaned_data.get('include_header')`, this returns cleaned/casted values 
1,true, True  for true values
0, false, False for false values

Reply:
Sounds good. Let's define a util function.

```python3
def str_to_bool(value):
    # https://github.com/django/django/blob/c765b62e3258de4dce9935ab7aed430346dfbc10/django/forms/fields.py#L790-L800
    if isinstance(value, str) and value.lower() in ("false", "0"):
        return False
    return bool(value)


# A unit test.
for value, expected in (
    ["True", True],
    ["true", True],
    ["1", True],
    ["False", False],
    ["false", False],
    ["0", False],
    ["", False],
):
    assert str_to_bool(value) == expected, f"{value} != {expected}"
```

Let's try to use filterset to filter out data instead of changing the response data structure.

Reply:
@thenav56  requested changes implemented